### PR TITLE
mysql: drop support for deprecated package & module option

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
- According to https://en.wikipedia.org/wiki/MySQL#Release_history MySQL 5.5 is past end of life so we shouldn't ship it anymore.
- The `services.mysql.rootPassword` option was never a good idea as it stores a password in the nix store. I'm straight up removing the option, as opposed to marking as deprecated, because this option completely breaks `services.mysql.ensureUsers` and `services.mysql.ensureDatabases`, which *many* NixOS services use now.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

ping @Izorkin @dasJ @dermetfan @thefloweringash @orivej

### CODE REVIEW

**Comment by Izorkin** [comment]
@aanderse can not delete mysql55 ? I use for one site, with mysql57 not worked.

**Comment by flokli** [comment]
@Izorkin We shouldn't keep around EOL MySQL versions in nixpkgs.

You should try to get that application working with more recent MySQL versions, and if that's absolutely not possible, you can pin an older version of nixpkgs for your specific usecase.

**Comment by flokli** [comment]
@GrahamcOfBorg build unixODBCDrivers.mysql
@GrahamcOfBorg test mysql mysqlbackup

File: pkgs/development/libraries/unixODBCDrivers/default.nix [hunk 1]
```diff
@@ -67,7 +67,7 @@
     };
 
     nativeBuildInputs = [ cmake ];
-    buildInputs = [ unixODBC mysql55 ];
+    buildInputs = [ unixODBC mysql ];
```
Comment (flokli) [inline pkgs/development/libraries/unixODBCDrivers/default.nix]: these derivations are marked as broken, and seem to fetch mysql 5.3 components even! Can we get these to work with a later mysql version, or maybe drop them too?

Comment (aanderse) [reply pkgs/development/libraries/unixODBCDrivers/default.nix]: It would be a real shame to drop instead of fix...

**Review by peterhoeg — APPROVED** [review]

**Review by fpletz — APPROVED** [review]

File: pkgs/development/libraries/unixODBCDrivers/default.nix (hunk 1 above)
Comment (aanderse) [reply pkgs/development/libraries/unixODBCDrivers/default.nix]: Maybe this will be fixed in a follow up PR.

**Pull request merged**